src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsNuGetFramework.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.VisualStudio
+{
+    public class VsNuGetFramework : IVsNuGetFramework
+    {
+        public VsNuGetFramework(
+            string targetFrameworkIdentifier,
+            string targetFrameworkVersion,
+            string profile,
+            string targetPlatformIdentifier,
+            string targetPlatformVersion)
+        {
+            TargetFrameworkIdentifier = targetFrameworkIdentifier ?? throw new ArgumentNullException(nameof(targetFrameworkIdentifier));
```

Comment:
This class, `VsNuGetFramework`, is basically just a COM-exposed version of `NuGet.Frameworks.NuGetFramework`, which never returns null values of properties, always empty strings. Also note that it's in a "private" assembly, so customers who want to pass in `IVsNuGetFramework` instances into `IVsFrameworkCompatibility3`'s methods need to implement the interface themselves, they can't use this implementation of that interface (to avoid the assembly versioning issue we avoid by using only COMImport interfaces with embed interop types.

Reply:
I buy that logic. 
However, in NuGetFramework we do accept nulls for certain cases, we just default to string.Empty.